src/plugins/connectionPlugin.ts

```diff
@@ -145,11 +143,6 @@ export type NodeValue<TypeName extends string = any, FieldName extends string =
 
 export type ConnectionFieldConfig<TypeName extends string = any, FieldName extends string = any> = {
   type: GetGen<'allOutputTypes', string> | AllNexusNamedOutputTypeDefs
-  /**
-   * Whether the connection field can be null
-   * @default (depends on whether nullability is configured in type or schema)
-   */
-  nullable?: boolean
```

Comment:
Before going into API design, I think you need to update the TS type of the `type` property to also accept `NexusNonNullDef<any> | GraphQLNonNull | ..`

Now, about that, we initially intentionally kept the `nullable: boolean` API on the connection plugin, because it felt wrong to express the nullability of the generated `Connection` type with wrappers. Example:

```ts
t.connectionField('booleanConnection', {
  type: nonNull('Boolean')
})
```
Will generate the following field:
```graphql
booleanConnection: BooleanConnection!
```

Using `nonNull('Boolean')` to generate `BooleanConnection!` feels a bit far-fetched, but I'll let you decide what you prefer

Reply:
Great point, I’m good with keeping the nullable api here! However, `t.nonNull.connectionField` should work too, right?

Reply:
Yes, fully agreed! `t.nonNull` should just set `nullable: false`